Octokit.Tests/Clients/RepositoryContentsClientTests.cs

```diff
@@ -89,5 +89,135 @@ public class TheGetContentsMethod
                 Assert.Equal(1, contents.Count);
             }
         }
+
+        public class TheCreateFileMethod
+        {
+            [Fact]
+            public void RequestsCorrectUrl()
+            {
+                var connection = Substitute.For<IApiConnection>();
+                var client = new RepositoryContentsClient(connection);
+
+                string expectedUri = "repos/org/repo/contents/path/to/file";
+                client.CreateFile("org", "repo", "path/to/file", new CreateFileRequest("message", "myfilecontents", "mybranch"));
+                
+                connection.Received().Put<RepositoryContentChangeSet>(Arg.Is<Uri>(u => u.ToString() == expectedUri), Arg.Any<object>());
```

Comment:
well the way i typically like to write unit tests is that each one has a specific focus/task to check.  EG this test is to check that the correct URL is used, therefore using the mocked out IApiConnection, it checks that the first parameter is a Uri that matches exactly what we expect to be called... but it's job isnt to check the 2nd parameter so it just ensures that it is "any object". 

Then there is another test that checks the 2nd parameter (PassesRequestObject) and in that one it doesnt care what the Uri was (just Arg.Any<Uri>) but it checks every field on the CreateFileRequest object was passed through.  So if you had a bug in the constructor where you swapped fields, or didnt populate the branch etc, that test would fail